### PR TITLE
Svg rendering fixes

### PR DESCRIPTION
**Pull request by frankban**
Make the bundle/diagram endpoint refer to the dynamic charm/icon one, so that icon rendering is faster and also the default icon is displayed when the charm has no icon.
Also fix the default icon rendering, avoiding new lines at the beginning of the SVG (XML declaration).

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/469/

File: internal/v4/defaulticon.go [hunk 1]
```diff
@@ -1,7 +1,9 @@
+// Copyright 2014 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
 package v4
 
-var defaultIcon = `
-<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+var defaultIcon = `<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (mhilton) [inline internal/v4/defaulticon.go:6]: Sorry I should have spotted that was a problem before.

**Comment by mhilton** [comment]
:+1:

File: internal/v4/content.go [hunk 2]
```diff
@@ -84,6 +85,7 @@ func (h *Handler) serveReadMe(id *charm.Reference, w http.ResponseWriter, req *h
 	return nil
 }
 
+// GET id/icon.svg
```
Comment (urosj) [inline internal/v4/content.go:89]: Please add: https://docs.google.com/a/canonical.com/document/d/1TgRA7jW_mmXoKH3JiwBbtPvQu7WiM6XMrz1wSrhTMXw/edit#bookmark=id.gv256equhhi6  - or short link :)

File: internal/v4/content.go [hunk 3]
```diff
@@ -64,6 +64,7 @@ var allowedReadMe = map[string]bool{
 	"readme.txt":      true,
 }
 
+// GET id/readme
```
Comment (urosj) [inline internal/v4/content.go:67]: Please add https://docs.google.com/a/canonical.com/document/d/1TgRA7jW_mmXoKH3JiwBbtPvQu7WiM6XMrz1wSrhTMXw/edit#bookmark=id.sn07dkq9ijht (or short link)

File: internal/v4/defaulticon_test.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	"github.com/juju/charmstore/internal/v4"
+)
+
+type iconSuite struct{}
+
+var _ = gc.Suite(&iconSuite{})
+
+func (s *iconSuite) TestValidXML(c *gc.C) {
+	// The XML declaration must be included in the first line of the icon.
+	hasXMLPrefix := strings.HasPrefix(v4.DefaultIcon, "<?xml")
```
Comment (urosj) [inline internal/v4/defaulticon_test.go:20]: I'd add a check that it actually includes substrings "<svg" and "/svg>" as well.

**Comment by urosj** [comment]
:+1: With suggestions. Thank you.

File: internal/v4/content.go (hunk 2 above)
Comment (frankban) [inline internal/v4/content.go:89]: Done.

File: internal/v4/content.go (hunk 3 above)
Comment (frankban) [inline internal/v4/content.go:67]: Done.

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/470/

**Comment by frankban** [comment]
Thank you both for the reviews!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/472/

**Pull request merged**